src/lib/replaceArticleCount.tsx

```diff
@@ -16,16 +16,16 @@ export const replaceArticleCount = (
     const parts = subbedText.split(/%%ARTICLE_COUNT_AND_NEXT_WORD%%/);
     const elements = [];
     for (let i = 0; i < parts.length - 1; i += 1) {
-        elements.push(<span dangerouslySetInnerHTML={{ __html: parts[i] }} />);
+        elements.push(<span dangerouslySetInnerHTML={{ __html: parts[i] as string }} />);
         elements.push(
             <ArticleCountOptOut
                 numArticles={numArticles}
-                nextWord={nextWords[i]}
+                nextWord={nextWords[i] as string}
                 type={articleCountOptOutType}
             />,
         );
     }
-    elements.push(<span dangerouslySetInnerHTML={{ __html: parts[parts.length - 1] }} />);
+    elements.push(<span dangerouslySetInnerHTML={{ __html: parts[parts.length - 1] as string }} />);
```

Comment:
All of these access are okay by inspection. `nextWords` is the same length as `parts`, the for loop is bound by the length of `parts` and the last access is to the final element of the array. It would be good if we could use a `forEach` loop here, but we need to iterate over two arrays :(. In python we could use the beautiful `zip` function: 

```python
for part, nextWord in zip(parts, nextWords):
    # ...
```

Reply:
You could use `map` to first zip the 2 arrays, and I guess it could also verify that `parts` and `nextWords` have the same length?
This function solves a tricky problem, and I cannot think of a nicer solution!